nuxeo-core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/api/Lock.java

```diff
@@ -77,4 +81,44 @@ public boolean getFailed() {
         return failed;
     }
 
+    /**
+     * @since 11.1
+     */
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == null) {
```

Comment:
I would prefer the simpler and easier to read version:
```java
return EqualsBuilder.reflectionEquals(this, obj);
```

Same for `toString`.